File: Scala/src/union_find/UnionFind.scala
package union_find

class UnionFind(vertexList: Array[Int]) {
    var parent: Array[Int] = new Array[Int](vertexList.length);

    for (i <- 0 until vertexList.length) {   //initialize parent array
        parent(i) = i;
    }

    def find(x: Int): Int = {
        if (parent(x) == x) {
            return x;
        }
        else {
            return find(parent(x));
        }
    }

    def union(x: Int, y: Int): Unit = {
        parent(find(y)) = find(x);
    }
}

File: Python/UnionFind.py
class UnionFind:
    def __init__(self, VertexList):
        # Initialize parent array
        self.parent = list(range(len(VertexList))) 

    def find(self, x):
        if self.parent[x] == x:
            return x
        else:
            self.parent[x] = self.find(self.parent[x])
            return self.parent[x]

    def union(self, x, y):
        self.parent[self.find(y)] = self.find(x)